crates/sgx_validation/src/config.rs

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize};
+use sgx::parameters::Features;
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// Values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary which are disallowed.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `features`.
```

Comment:
`Features` is defined & documented in the SGX crate. How many places need to have documentation for the same thing? Does it make sense to copy some/all of the documentation here since end-users might be interacting with SGX concepts?

Reply:
This is a completely opaque `u64`
`features` does not mean anything on it's own.

If this represents https://docs.rs/sgx/latest/sgx/parameters/struct.Features.html, then I'd expect this field to be of type `Features` (https://docs.rs/sgx/latest/sgx/parameters/struct.Features.html). If it's `Features::is_empty`(https://docs.rs/sgx/latest/sgx/parameters/struct.Features.html#method.is_empty), then it's "not set"

If the type is in the type signature, not much needs to be done, although you still need to explain what is the semantic value of this field. What happens if it's not set? What happens if it is set?

Reply:
Still not addressed

Reply:
I believe this is resolved

Reply:
Not, it is not. There's no documentation for this field.

Reply:
`Features` is now the features struct from the SGX crate, so no longer opaque. Still have to have an intermediate Enum since the SGX Features doesn't implement `deserialize`.